crates/flox-rust-sdk/src/models/environment/managed_environment.rs

```diff
@@ -836,4 +869,75 @@ mod test {
         ManagedEnvironment::ensure_branch("branch_2", &lock, &floxmeta).unwrap();
         assert_eq!(floxmeta.git.branch_hash("branch_2").unwrap(), hash_1);
     }
+
+    #[test]
+    fn stable_encode_name() {
+        let path = PathBuf::from_str("foo").unwrap();
+        let encode1 = ManagedEnvironment::encode(&path);
+        let encode2 = ManagedEnvironment::encode(&path);
+        assert_eq!(encode1, encode2);
```

Comment:
suggestion: check some cases like `dir/../dir` vs `dir` and `dir` vs `/abs/path/to/dir`